skyportal/handlers/api/allocation.py

```diff
@@ -640,21 +640,25 @@ def put(self, allocation_id):
 
             data = self.get_json()
             data["id"] = allocation_id
+
+            replace_altdata = data.pop("replace_altdata", False)
             if isinstance(data.get("_altdata"), dict):
-                # sanitize the dictionnary, removing keys with null or empty values
+                # sanitize the dictionary, removing keys with null or empty values
                 data["_altdata"] = {
                     k: v
                     for k, v in data["_altdata"].items()
                     if v is not None and str(v).strip() != ""
                 }
-                # if it ends up being a dictionnary with no keys, remove it
-                # otherwise convert it to a string
-                if not data["_altdata"] or not any(data["_altdata"]):
+
+                if not data["_altdata"]:
                     data.pop("_altdata")
+                elif allocation._altdata and replace_altdata:
```

Comment:
hmmmm doesn't that feel somewhat reversed? Idk but to me it feels like "replace altdata" means fully replacing it, not updating its content with newly provided values if any. Idk I feel like the naming here is a bit confusing, any thoughts @mcoughlin?